tools/test.py

```diff
@@ -3,127 +3,120 @@
 
 
 def main(argv):
-    if len(argv) == 2:
-        build_dir = sys.argv[1]
-    elif len(argv) == 1:
-        build_dir = build_path()
-    else:
-        print "Usage: tools/test.py [build_dir]"
-        sys.exit(1)
-
-    deno_dir = os.path.join(build_dir, ".deno_test")
+    args = test_args(argv)
+
+    deno_dir = os.path.join(args.build_dir, ".deno_test")
     if os.path.isdir(deno_dir):
         rmtree(deno_dir)
     os.environ["DENO_DIR"] = deno_dir
 
     enable_ansi_colors()
 
-    http_server.spawn()
-
-    deno_exe = os.path.join(build_dir, "deno" + executable_suffix)
-    check_exists(deno_exe)
-
-    # Python/build tools testing
-    setup_test()
-    util_test()
-
-    run([
-        "node", "./node_modules/.bin/ts-node", "--project",
-        "tools/ts_library_builder/tsconfig.json",
-        "tools/ts_library_builder/test.ts"
-    ])
-
-    libdeno_test = os.path.join(build_dir, "libdeno_test" + executable_suffix)
-    check_exists(libdeno_test)
-    run([libdeno_test])
-
-    cli_test = os.path.join(build_dir, "cli_test" + executable_suffix)
-    check_exists(cli_test)
-    run([cli_test])
-
-    deno_core_test = os.path.join(build_dir,
-                                  "deno_core_test" + executable_suffix)
-    check_exists(deno_core_test)
-    run([deno_core_test])
-
-    deno_core_http_bench_test = os.path.join(
-        build_dir, "deno_core_http_bench_test" + executable_suffix)
-    check_exists(deno_core_http_bench_test)
-    run([deno_core_http_bench_test])
-
-    unit_tests(deno_exe)
-
-    fetch_test(deno_exe)
-    fmt_test(deno_exe)
-
-    integration_tests(deno_exe)
-
-    # TODO We currently skip testing the prompt and IsTTY in Windows completely.
-    # Windows does not support the pty module used for testing the permission
-    # prompt.
-    if os.name != 'nt':
-        from is_tty_test import is_tty_test
-        from permission_prompt_test import permission_prompt_test
-        from complex_permissions_test import complex_permissions_test
-        permission_prompt_test(deno_exe)
-        complex_permissions_test(deno_exe)
-        is_tty_test(deno_exe)
-
-    repl_tests(deno_exe)
-
-    rmtree(deno_dir)
-
-    deno_dir_test(deno_exe, deno_dir)
-
-    test_no_color(deno_exe)
-
-    benchmark_test(build_dir, deno_exe)
-    exec_path_test(deno_exe)
+    with Spawn():
+        test_cases = [
+            TestSetup,
+            TestUtil,
+            TestLib,
+            JsUnitTests,
+            FetchTest,
+            FmtTest,
+            TestIntegrations,
+            TestRepl,
+            TestDenoDir,
+            TestBenchmark,
+        ]
+        if os.name != 'nt':
+            test_cases.append(TestIsTty)
+            test_cases += permission_prompt_tests()
+            test_cases += complex_permissions_tests()
+
+        suite = unittest.TestSuite([
+            unittest.TestLoader().loadTestsFromTestCase(tc)
+            for tc in test_cases
+        ])
+
+        result = ColorTextTestRunner(
```

Comment:
Replace with `test_main`

Reply:
I'm not sure it looks up suite correctly, can you pass suite to main?

Reply:
I mean `test_main` that you wrote

Reply:
You might be right, this might not be possible

Reply:
I think I want to not call main in test_main, but rather do the module lookup there (unless suite is passed), I don't think this needs to be done yet. Benefit is simplified arg parsing (main parses args too).